test/parallel/test-stdio-undestroy.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const spawn = require('child_process').spawn;
+const fs = require('fs');
+const fixtures = require('../common/fixtures');
+
+if (process.argv[2] === 'child') {
+  process.stdout.destroy()
+  console.log('out')
+  console.error('err')
```

Comment:
We should also test that 
- This applies when `process.stderr.destroy()` is also called (otherwise the `console.error` test is not meaningful)
- Explicitly using `process.stdout.write()` works
- Explicitly using `process.stderr.write()` works
- Writing after a short timeout works (since `destroy()` can, generally speaking, be asynchronous)